bundles/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/handler/ExecHandler.java

```diff
@@ -160,21 +172,61 @@ public void run() {
                     }
                 } catch (IllegalFormatException e) {
-                    logger.error(
+                    logger.warn(
                             "An exception occurred while formatting the command line with the current time and input values : '{}'",
                             e.getMessage());
                     updateState(RUN, OnOffType.OFF);
                     return;
                 }
 
-                logger.trace("The command to be executed will be '{}'", commandLine);
+                String[] cmdArray;
+                String[] shell;
+                if (commandLine.contains(CMD_LINE_DELIMITER)) {
+                    logger.debug("Splitting by '{}'", CMD_LINE_DELIMITER);
+                    try {
+                        cmdArray = commandLine.split(CMD_LINE_DELIMITER);
+                    } catch (PatternSyntaxException e) {
+                        logger.warn("An exception occurred while splitting '{}' : '{}'",
+                                commandLine, e.getMessage());
+                        updateState(RUN, OnOffType.OFF);
+                        updateState(OUTPUT, new StringType(e.getMessage()));
+                        return;
+                    }
+                } else {
+                    // Invoke shell with 'c' option and pass string
+                    logger.debug("Passing to shell for parsing command.");
+                    if (getOperatingSystemType() == OS.WINDOWS) {
+                        shell = SHELL_WINDOWS;
+                        logger.debug("OS: WINDOWS ({})", getOperatingSystemName());
+                        cmdArray = createCmdArray(shell, "/c", commandLine);
+                    } else if (getOperatingSystemType() != OS.UNKNOWN) {
+                        // assume sh is present, should all be POSIX-compliant
+                        shell = SHELL_NIX;
+                        logger.debug("OS: *NIX ({})", getOperatingSystemName());
+                        cmdArray = createCmdArray(shell, "-c", commandLine);
+                    } else {
+                        String err = "OS not supported, please manually split commands!";
+                        logger.debug("OS: Unknown ({})", getOperatingSystemName());
+                        logger.warn(err);
+                        updateState(RUN, OnOffType.OFF);
+                        updateState(OUTPUT, new StringType(err));
+                        return;
+                    }
+                }
+
+                if (cmdArray.length == 0) {
+                    logger.trace("Empty command received, not executing");
+                    return;
+                }
+
+                logger.trace("The command to be executed will be '{}'", Arrays.asList(cmdArray));
 
                 Process proc = null;
                 try {
-                    proc = rt.exec(commandLine.toString());
+                    proc = rt.exec(cmdArray);
                 } catch (Exception e) {
```

Comment:
Is it necessary to catch `Exception` here? According to [the docs](https://docs.oracle.com/javase/7/docs/api/java/lang/Runtime.html#exec(java.lang.String[])) there is a limited number of exceptions, I think we can omit the NPE if we make sure that none of the commands is null (which should be done anyway).